packages/plugin-snowplow/src/androidMain/kotlin/ly.iterative.itly.snowplow/SnowplowPlugin.kt

```diff
@@ -0,0 +1,67 @@
+/**
+ * Android SnowplowPlugin
+ */
+package ly.iterative.itly.snowplow
+
+import ly.iterative.itly.*
+
+import com.snowplowanalytics.snowplow.Snowplow;
+import com.snowplowanalytics.snowplow.controller.TrackerController
+import com.snowplowanalytics.snowplow.event.SelfDescribing
+import com.snowplowanalytics.snowplow.network.HttpMethod
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+
+open class SnowplowCallOptions : PluginCallOptions()
+class SnowplowAliasOptions : SnowplowCallOptions()
+class SnowplowGroupOptions : SnowplowCallOptions()
+class SnowplowIdentifyOptions : SnowplowCallOptions()
+class SnowplowTrackOptions constructor(
+        callback: (() -> Unit)?,
+        context: MutableList<SelfDescribingJson>?
+) : SnowplowCallOptions() {
+    var callback: (() -> Unit)? = callback
+    var context: MutableList<SelfDescribingJson>? = context
+}
+
+actual class SnowplowPlugin actual constructor(
+    options: SnowplowOptions
+) : Plugin<SnowplowAliasOptions, SnowplowIdentifyOptions, SnowplowGroupOptions, SnowplowTrackOptions>(ID) {
+    companion object {
+        @JvmField
+        val ID = "snowplow"
+    }
+
+    private var config: SnowplowOptions = options
+    private lateinit var logger: Logger
+    private lateinit var snowplow: TrackerController
+
+    val client: TrackerController?
+        get() = this.snowplow
+
+    override fun load(options: PluginLoadOptions) {
+        logger = options.logger
+        logger.debug("[plugin-snowplow] load")
+        this.snowplow = config.tracker ?: Snowplow.createTracker(config.androidContext,
+                "itly", config.trackerUrl, HttpMethod.POST);
```

Comment:
@dantetam now that tracker and trackerUrl are both optional please throw an error if neither are set. Right now this line has a build error.